### DISCUSSION TITLE
how to set cpp indent as 4 space when save files?

### DISCUSSION THREAD

**Discussion by edte** [nvim-treesitter/nvim-treesitter#3175] [Q&A]
the default indent is 2 space,

![a](https://user-images.githubusercontent.com/50194671/177959638-b8dd48c8-28e5-43dc-8b79-5a8bc345d671.gif)

And how to set the number of words for automatic line wrapping? It's too early to wrap

**Comment by theHamsta** [comment]
`:set shiftwidth=4` if I remember correctly. I the automatic wrapping happens through formatting with clang-format through your language server, you have to adapt `.clang-format` in the project root. This project does no automatic formatting and wrapping for display by neovim only happens at end of line

**Reply by edte** [reply]
> `:set shiftwidth=4` if I remember correctly. I the automatic wrapping happens through formatting with clang-format through your language server, you have to adapt `.clang-format` in the project root. This project does no automatic formatting and wrapping for display by neovim only happens at end of line

Formatting with calngd can be configured like this, but when using ctrl+s to save the file and automatically format it, even when using the format of nvim-lsp, the format will still be 2 space
